Miracast/common/MiracastCommon.cpp

```diff
@@ -0,0 +1,193 @@
+        // Join thread
+        pthread_join(m_pthread_id, nullptr);
+        m_pthread_id = 0;
+        pthread_attr_destroy(&m_pthread_attr);
+    }
+
+    if ( nullptr != m_g_queue ){
+        // Close message queue
+        g_async_queue_unref(m_g_queue);
+        sem_destroy(&m_empty_msgq_sem_obj);
+        m_g_queue = nullptr;
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+}
+
+MiracastError MiracastThread::start(void)
+{
+    MiracastError ret_code = MIRACAST_OK;
+    MIRACASTLOG_TRACE("Entering...");
+    if ( 0 != pthread_create(   &m_pthread_id,
+                                &m_pthread_attr,
+                                reinterpret_cast<void *(*)(void *)>(m_thread_callback),
+                                m_thread_user_data))
+    {
+        ret_code = MIRACAST_FAIL;
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return ret_code;
+}
+
+void MiracastThread::send_message(void *message, size_t msg_size)
+{
+    MIRACASTLOG_TRACE("Entering...");
+    if (nullptr != m_g_queue){
+        void *buffer = malloc(msg_size);
+        if (nullptr == buffer)
+        {
+            MIRACASTLOG_ERROR("Memory Allocation Failed for %u\n", msg_size);
+            MIRACASTLOG_TRACE("Exiting...");
+            return;
+        }
+        memset(buffer, 0x00, msg_size);
+        // Send message to queue
+
+        memcpy(buffer, message, msg_size);
+        g_async_queue_push(m_g_queue, buffer);
+        sem_post(&m_empty_msgq_sem_obj);
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+}
+
+int8_t MiracastThread::receive_message(void *message, size_t msg_size, int sem_wait_timedout)
+{
+    int8_t status = false;
+    MIRACASTLOG_TRACE("Entering...");
+    if (nullptr != m_g_queue){
+        if (( THREAD_RECV_MSG_WAIT_IMMEDIATE == sem_wait_timedout )||
+            (THREAD_RECV_MSG_INDEFINITE_WAIT == sem_wait_timedout))
+        {
+            int count = 0;
+	        sem_getvalue(&m_empty_msgq_sem_obj,&count);
+            if ((0 < count ) || (THREAD_RECV_MSG_INDEFINITE_WAIT == sem_wait_timedout)){
+                sem_wait(&m_empty_msgq_sem_obj);
+                status = true;
+            }
+        }
+        else if (0 < sem_wait_timedout)
+        {
+            struct timespec ts;
+            clock_gettime(CLOCK_REALTIME, &ts);
+            ts.tv_sec += sem_wait_timedout;
+
+            if (-1 != sem_timedwait(&m_empty_msgq_sem_obj, &ts))
+            {
+                status = true;
+            }
+        }
+        else
+        {
+            status = -1;
+        }
+
+        if (true == status)
+        {
+            void *data_ptr = static_cast<void *>(g_async_queue_pop(m_g_queue));
+            if ((nullptr != message) && (nullptr != data_ptr))
+            {
+                memcpy(message, data_ptr, msg_size);
+                free(data_ptr);
+            }
+        }
+    }
+    MIRACASTLOG_TRACE("Exiting...");
+    return status;
+}
+
+#if 0
+std::string parse_opt_flag( std::string file_name , bool integer_check )
```

Comment:
do we  need this

Reply:
This API generic to read the opt flags. So we can keep it in this common file. Later we can use it